File: forgerock-auth/src/main/java/org/forgerock/android/auth/callback/TermsAndConditionsCallback.java
/*
 * Copyright (c) 2019 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package org.forgerock.android.auth.callback;

import androidx.annotation.Keep;

import org.json.JSONException;
import org.json.JSONObject;

import lombok.Getter;

/**
 * A callback to collect a user's acceptance of the configured Terms &amp; Conditions.
 */
@Getter
public class TermsAndConditionsCallback extends AbstractCallback {

    /**
     * Retrieves the version.
     *
     * @return the version
     */
    private String version;

    /**
     * Retrieve the Terms &amp; Conditions.
     *
     * @return the Terms &amp; Conditions
     */
    private String terms;

    /**
     * The create date of the Terms &amp; Conditions.
     *
     * @return the create date.
     */
    private String createDate;

    @Keep
    public TermsAndConditionsCallback(JSONObject raw, int index) throws JSONException {
        super(raw, index);
    }

    @Keep
    public TermsAndConditionsCallback() {
    }

    @Override
    protected void setAttribute(String name, Object value) {
        switch (name) {
            case "version":
                this.version = (String) value;
                break;
            case "terms":
                this.terms = (String) value;
                break;
            case "createDate":
                this.createDate = (String) value;
                break;

            default:
                //ignore
        }
    }

    /**
     * Sets whether the Terms have been accepted.
     *
     * @param accept boolean representing the acceptance of the terms.
     */
    public void setAccept(boolean accept) {
        setValue(accept);
    }

    @Override
    public String getType() {
        return "TermsAndConditionsCallback";
    }
}

File: e2e/app/src/main/java/com/example/app/callback/TermsAndConditionsCallback.kt
/*
 * Copyright (c) 2024 ForgeRock. All rights reserved.
 *
 *  This software may be modified and distributed under the terms
 *  of the MIT license. See the LICENSE file for details.
 */

package com.example.app.callback

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import org.forgerock.android.auth.callback.TermsAndConditionsCallback

@Composable
fun TermsAndConditionsCallback(callback: TermsAndConditionsCallback) {

    var input by remember {
        mutableStateOf(false)
    }

    Row(modifier = Modifier
        .padding(16.dp)
        .fillMaxWidth()) {
        Text(text = callback.version,
            Modifier
                .weight(1f),
            style = MaterialTheme.typography.titleMedium
        )
        Spacer(Modifier.width(8.dp))
        Text(text = callback.createDate,
            Modifier
                .weight(1f),
            style = MaterialTheme.typography.titleMedium
        )
        Spacer(Modifier.width(8.dp))
        Text(text = callback.terms,
            Modifier
                .weight(1f),
            style = MaterialTheme.typography.titleSmall
        )
        Spacer(Modifier.width(8.dp))
        Switch(
            checked = input,
            onCheckedChange = {
                input = it
                callback.setAccept(it)
            }
        )
    }

}